Public/Src/Engine/Processes/FileAccessManifest.cs

```diff
@@ -448,7 +457,31 @@ public void AddPath(AbsolutePath path, FileAccessPolicy mask, FileAccessPolicy v
 
             m_rootNode.AddNodeWithScope(this, path, new FileAccessScope(mask, values), expectedUsn ?? ReportedFileAccess.NoUsn);
         }
+
+        private void WriteAnyBuildShimBlock(BinaryWriter writer)
+        {
+#if DEBUG
+            writer.Write(0xABCDEF04); // "ABCDEF04"
+#endif
+            WriteChars(writer,

```

Comment:
>W [](start = 12, length = 1)

Why not have the following condition?

```
if (AnyBuildShimInfo != null) { writer.Write(1); /* Do the stuff you have there */ }
else { writer.Write(0); }
```

At least it reduces the occurrences of `?.` and `??` operators.